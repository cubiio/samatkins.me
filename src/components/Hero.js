import React from 'react';
import styled from 'styled-components';
import HeroLists from '../components/HeroLists';
import { NAV_ANCHOR, NAV_ANCHOR_HOVER } from '../lib/theme/colours';

const Hero = () => (
  <HeroWrapper>
    <HeroIntro>
      <br />
      <p>
        Hi, I&#39;m Sam, welcome to my site! I&#39;m a Software Engineer at{' '}
        <a href="https://www.bighealth.com/">Big Health</a> and I&#39;m enjoying
        doing work that matters. I work across front and back-end technologies
        including React, React Native, and Python.
      </p>
      <p>
        My background is in supply chain and project management, and I worked
        for over nine years in Big Four management consulting.
      </p>
      <p>
        Outside of work, I&#39;m a family man and I like cycling, mountain
        biking, football and rugby.
      </p>
    </HeroIntro>
    <HeroLists />
  </HeroWrapper>
);

export default Hero;

const HeroWrapper = styled.div`
  display: flex;
  flex-direction: column;
  width: 50%;
  margin-left: auto;
  margin-right: auto;

  @media (max-width: 700px) {
    width: 95%;
    text-align: center;
  }
`;

const HeroIntro = styled.div`
  margin-bottom: 30px;
  margin-left: auto;
  margin-right: auto;

  @media (max-width: 700px) {
    width: 95%;
  }

  a {
    color: ${NAV_ANCHOR};
    text-decoration: none;
  }

  a:hover {
    color: ${NAV_ANCHOR_HOVER};
  }
`;
