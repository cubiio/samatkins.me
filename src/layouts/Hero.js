import React, { Component } from 'react'
import SkillsOverview from '../layouts/SkillsOverview'

class Hero extends Component {
  render () {
    return (
      <div className='hero__wrapper'>
        <h2 className='hero__tagline'>Dad. Husband. Web developer.</h2>
        <div className='hero__intro'>
          <p>Iâ€™m a Junior Full-Stack Developer at <a href='https://www.bighealth.com/'>Big Health</a>.</p>
        </div>
        <SkillsOverview />
      </div>
    )
  }
}

export default Hero
