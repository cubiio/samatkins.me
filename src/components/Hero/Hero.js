import React from 'react';
import HeroLists from '../HeroLists';
import * as Styles from './Hero.styles';

const Hero = () => (
  <Styles.HeroWrapper>
    <Styles.HeroIntro>
      <br />
      <p>
        Hi, I&#39;m Sam, welcome to my site! I&#39;m a Software Engineer at{' '}
        <a href="https://www.bighealth.com/">Big Health</a> and I&#39;m enjoying
        doing work that matters. I work across front and back-end technologies
        including React, React Native, and Python.
      </p>
      <p>
        My background is in supply chain and project management, and I worked
        for over nine years in Big Four management consulting.
      </p>
      <p>
        Outside of work, I&#39;m a family man and I like cycling, mountain
        biking, football and rugby.
      </p>
    </Styles.HeroIntro>
    <HeroLists />
  </Styles.HeroWrapper>
);

export default Hero;
