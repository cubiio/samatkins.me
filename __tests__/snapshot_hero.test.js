/* global it expect */

import React from 'react'
import Hero from '../src/components/Hero'
import renderer from 'react-test-renderer'

it('renders correctly', () => {
  const tree = renderer.create(
    <Hero />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})
