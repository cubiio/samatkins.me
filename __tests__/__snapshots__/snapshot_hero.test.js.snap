// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="sc-EHOje cMrZsj"
>
  <h2
    className="sc-bZQynM kuZEQm"
  >
    Dad. Husband. Web developer.
  </h2>
  <div
    className="sc-gzVnrw fyGwWZ"
  >
    <p>
      I live in London with my family, and I work at 
      <a
        href="https://www.bighealth.com/"
      >
        Big Health
      </a>
       as a Junior Full-Stack Developer. I work across front-end and back-end technologies but my passion is front-end, and I currently work and play around with React and React Native.
    </p>
    <p>
      My background is in supply chain and project management, and I worked for over nine years in Big Four management consulting.
    </p>
    <p>
      Outside of work, Iâ€™m a family man and I like cycling, mountain biking, football and rugby.
    </p>
    <p>
      This is my personal website, portfolio, and blog. I also list the technologies I use and want to learn below. Apologies for the lack of design on the site, this is due to time constraints but mainly because whilst I want to learn more about good design, I will always be a developer rather than a designer.
    </p>
  </div>
  <div
    className="sc-htpNat iqWsRc"
  >
    <h2
      className="sc-bxivhb cjdjhR"
    >
      My tech stack:
    </h2>
    <div
      className="sc-ifAKCX iurdrm"
    >
      <div>
        <div
          className="sc-bdVaJa ipmtpW"
        >
          <h3
            className="sc-bwzfXH iAdRya"
          >
            Core:
          </h3>
          <ul>
            <p>
              - 
              JavaScript
            </p>
            <p>
              - 
              Python
            </p>
            <p>
              - 
              HTML
            </p>
            <p>
              - 
              CSS
            </p>
          </ul>
        </div>
      </div>
      <div>
        <div
          className="sc-bdVaJa ipmtpW"
        >
          <h3
            className="sc-bwzfXH iAdRya"
          >
            Libraries:
          </h3>
          <ul>
            <p>
              - 
              React
            </p>
            <p>
              - 
              React Native
            </p>
            <p>
              - 
              Python Flask
            </p>
          </ul>
        </div>
      </div>
      <div>
        <div
          className="sc-bdVaJa ipmtpW"
        >
          <h3
            className="sc-bwzfXH iAdRya"
          >
            Current dev stack:
          </h3>
          <ul>
            <p>
              - 
              JavaScript (ES6)
            </p>
            <p>
              - 
              React / React Native
            </p>
            <p>
              - 
              Python / Python Flask
            </p>
            <p>
              - 
              Docker
            </p>
            <p>
              - 
              Gatsby JS
            </p>
          </ul>
        </div>
      </div>
      <div>
        <div
          className="sc-bdVaJa ipmtpW"
        >
          <h3
            className="sc-bwzfXH iAdRya"
          >
            Tools:
          </h3>
          <ul>
            <p>
              - 
              Atom.io
            </p>
            <p>
              - 
              iTerm2
            </p>
            <p>
              - 
              Sourcetree
            </p>
            <p>
              - 
              Xcode
            </p>
            <p>
              - 
              Bear-Writer.com
            </p>
            <p>
              - 
              Dash Docs
            </p>
          </ul>
        </div>
      </div>
      <div>
        <div
          className="sc-bdVaJa ipmtpW"
        >
          <h3
            className="sc-bwzfXH iAdRya"
          >
            Learning list:
          </h3>
          <ul>
            <p>
              - 
              JavaScript (ES6 / 7)
            </p>
            <p>
              - 
              React / React Native
            </p>
            <p>
              - 
              Mobile development
            </p>
            <p>
              - 
              Jest tests
            </p>
            <p>
              - 
              Flow
            </p>
            <p>
              - 
              Webpack
            </p>
          </ul>
        </div>
      </div>
      <div>
        <div
          className="sc-bdVaJa ipmtpW"
        >
          <h3
            className="sc-bwzfXH iAdRya"
          >
            Experimented with:
          </h3>
          <ul>
            <p>
              - 
              PHP
            </p>
            <p>
              - 
              Objective C
            </p>
            <p>
              - 
              Sass
            </p>
            <p>
              - 
              Node.js
            </p>
            <p>
              - 
              Knockout.js
            </p>
            <p>
              - 
              Gulp
            </p>
          </ul>
        </div>
      </div>
    </div>
  </div>
</div>
`;
